app/pages/Account/functions.js

```diff
@@ -0,0 +1,313 @@
+import walletClient from "../../utils/walletClient";
+import * as networks from "hsd/lib/protocol/networks";
+
+const ZERO_HASH = Buffer.alloc(32, 0x00).toString("hex");
+
+function isExpired(auction, height, network) {
+  if (auction.revoked !== 0) {
+    if (height < auction.revoked + network.names.auctionMaturity) return false;
+    return true;
+  }
+
+  // Can only renew once we reach the closed state.
+  if (auction.state !== "CLOSED") {
+    return false;
+  }
+
+  // Claimed names can only expire once the claim period is over.
+  // if (this.isClaimable(height, network))
+  if (
+    auction.claimed !== 0 &&
+    !network.names.noReserved &&
+    height < network.names.claimPeriod
+  ) {
+    return false;
+  }
+
+  // If we haven't been renewed in a year, start over.
+  if (height >= auction.renewal + network.names.renewalWindow) {
+    return true;
+  }
+
+  // If nobody revealed their bids, start over.
+  // if (auction.owner.isNull())
+  if (auction.owner.hash === ZERO_HASH && auction.owner.index === 0xffffffff) {
+    return true;
+  }
+
+  return false;
+}
+
+async function getStatsFromBids() {
```

Comment:
i think we should move `getStatsFromBids` and `getStatsFromName` to background, and on UI thread we can just call `walletClient.getBidStats` and `walletClient.getNameStats`